Document dbt system command (#9493)
fishtown-analytics/docs.getdbt.com: website/docs/reference/commands/system.md

---
title: "About dbt system command"
sidebar_label: "system"
id: "system"
description: "Use dbt system to update or uninstall the Fusion engine using the CLI, or pre-install supported ADBC adapter drivers."
---

# About dbt system command

> `dbt system` provides commands for managing the <Constant name="fusion" /> CLI installation. The <Constant name="fusion_engine" /> is a single compiled binary with no dependencies on other libraries. Once installed, `dbt system` lets you update, uninstall, and manage drivers directly

Reviewer: Either this or a period.
```suggestion
`dbt system` provides commands for managing the <Constant name="fusion" /> CLI installation. The <Constant name="fusion_engine" /> is a single compiled binary with no dependencies on other libraries. Once installed, `dbt system` lets you update, uninstall, and manage drivers directly without pip, brew, or a Python environment. 
```

`dbt system update` and `dbt system uninstall` are only supported for [CDN installations](/docs/local/install-dbt) of <Constant name="fusion" />. If you installed <Constant name="fusion" /> via Pip, Homebrew, or Winget, you'll need to update or uninstall dbt from those tools.

<Constant name="core" /> has no equivalent.

## Usage

```shell
dbt system <subcommand> [flags]
```

## Subcommands

| Subcommand | Description |
|---|---|

> | `dbt system update` | Update CLI to the latest (or a specified) version |

Reviewer: should we change these to say "fusion engine" as well?

> | `dbt system update` | Update CLI to the latest (or a specified) version |

Author: ```suggestion
| `dbt system update` | Update <Constant name="fusion" /> CLI to the latest (or a specified) version |
```

> | `dbt system uninstall` | Remove CLI from your system |

Author: ```suggestion
| `dbt system uninstall` | Remove <Constant name="fusion" />  from your system |
```

---
title: "About dbt system command"
sidebar_label: "system"
id: "system"

> description: "Use dbt system to update, uninstall, or pre-install ADBC adapter drivers for the dbt CLI."

Reviewer: It read like you were updating/uninstalling the drivers. Seems like according to Elias' comment, we might want to reframe dbt CLI > Fusion?
```suggestion
description: "Use dbt system to update or uninstall the Fusion engine using the CLI, or pre-install supported ADBC adapter drivers."
```

---

# About dbt system command

> `dbt system` provides commands for managing the <Constant name="fusion" /> CLI and <Constant name="platform_cli" /> installations. <Constant name="core" /> has no equivalent.

Reviewer 2: I don't think it is also functional in context of the platform CLI. I think it's Fusion specific.

> `dbt system` provides commands for managing the <Constant name="fusion" /> CLI and <Constant name="platform_cli" /> installations. <Constant name="core" /> has no equivalent.

Reviewer: @matthewshaver can we add more context to why this is helpful? Maybe something like:

> The dbt Fusion engine is a single compiled binary with no dependencies on other libraries. Once installed, dbt system lets you update, uninstall, and manage drivers directly without pip, brew, or a Python environment. You can use the subcommands on macOS, Linux, and Windows.

> `dbt system` provides commands for managing the <Constant name="fusion" /> CLI and <Constant name="platform_cli" /> installations. <Constant name="core" /> has no equivalent.

Author: ```suggestion
`dbt system` provides commands for managing the <Constant name="fusion" /> CLI installation. <Constant name="core" /> has no equivalent.
```

## Usage

```shell
dbt system <subcommand> [flags]
```

## Subcommands

| Subcommand | Description |
|---|---|
| `dbt system update` | Update CLI to the latest (or a specified) version |
| `dbt system uninstall` | Remove CLI from your system |
| `dbt system install-drivers` | Pre-install all supported ADBC adapter drivers |

## dbt system update

`dbt system update` updates the CLI binary in place. By default, it updates to the latest stable release. You can also target a named channel or a specific version number.

```shell
dbt system update [flags]
```

### Options

| Flag | Description |
|---|---|

> | `--version VERSION` | Update to a specific version (for example, `1.2.3`) or a named channel (`canary`, `dev`). Defaults to the latest stable release. |

Reviewer 2: Can we make this example more clear? i.e. `2.0.0-preview.190` then in the future `2.0.2`.

> | `--version VERSION` | Update to a specific version (for example, `1.2.3`) or a named channel (`canary`, `dev`). Defaults to the latest stable release. |

Author: ```suggestion
| `--version VERSION` | Update to a specific version (for example, `2.0.0-preview.190`) or a named channel (`canary`, `dev`). Defaults to the latest stable release. |
```

> | `--package PACKAGE` | Which package to update. Accepts `dbt`, `dbt-lsp`, or `all`. Defaults to `dbt`. |

Reviewer 2: dbt-lsp is no longer a separate binary so this flag is irrelevant. I wouldn't include it in docs

> | `--package PACKAGE` | Which package to update. Accepts `dbt`, `dbt-lsp`, or `all`. Defaults to `dbt`. |

Author: ```suggestion
| `--package PACKAGE` | Which package to update. Accepts `dbt` or `all`. Defaults to `dbt`. |
```

### Version channels

| Channel | Description |
|---|---|
| _(none)_ | Latest stable release |

> | `canary` | Most recent preview build |

Reviewer 2: Worth noting that canary goes through active testing and can be pulled back. That's the point of canary.

> | `canary` | Most recent preview build |

Author: ```suggestion
| `canary` | Most recent preview build (under active testing and may be pulled back) |
```

| `dev` | Development build (may be unstable) |

### Examples

Update to the latest stable release:

```shell
dbt system update
```

Update to a specific version:

```shell
dbt system update --version 2.0.0-preview.188
```

Update to the canary channel:

```shell
dbt system update --version canary
```

> Update both `dbt` and `dbt-lsp` to the latest version:

Reviewer 2: Also should remove this since dbt-lsp is gone

> Update both `dbt` and `dbt-lsp` to the latest version:

Author: ```suggestion
```
